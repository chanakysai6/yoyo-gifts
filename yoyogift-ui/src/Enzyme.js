import Enzyme, { configure, shallow, mount, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
const cardDetails = {
  id: 1,
  cardName: "Amazon Gift ",
  cardPoints: "1200",
  cardCategory: "Ecommerce",
  cardRetailer: "Amazon",
  cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
  cardExpiryDate: "2019-06-29T00:00:00.000Z",
  cardCount: 95,
  cardImage: "https://pngimg.com/uploads/amazon/amazon_PNG21.png",
  cardVendor: "amazon",
  cardShortDesc: "10% OFF",
  receiverEmail: "test.mail.com",
  userEmail: "test.mail.com",
  cardLongDesc:
    "Amazon Gift Cards are the Perfect Gift, Every Time. Use the eBay Gift Card to shop from millions of items in Electronics, Toys, Motors, Fashion, Home & Garden, Art, Collectibles, Sporting Goods and everything in-between. eBay Gift Cards never expire and have no fees. Use it to shop now or wait for the deal of a lifetime.",
  cardComments: [
    {
      id: 1,
      first_name: "Sebastian",
      last_name: "Eschweiler",
      email: "sebastian@mindtree.com",
      rating: 4,
      comment: "Great gift card. Happy to gift!",
      commented_on: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
    },
    {
      id: 2,
      first_name: "Bill",
      last_name: "Joe",
      email: "billjoe@mindtree.com",
      rating: 3,
      comment: "10% discount on Amazon. Yay",
      commented_on: "Sun Jun 12 2019 15:43:25 GMT+0530 (India Standard Time)"
    },
    {
      id: 3,
      first_name: "Murphy",
      last_name: "Dil",
      email: "murphydil@mindtree.com",
      rating: 3,
      comment: "Woah! Instant discount.",
      commented_on: "Sun Apr 20 2019 15:43:25 GMT+0530 (India Standard Time)"
    }
  ]
};
const userDetails = {
  id: "115728151885118053689",
  email: "soumodips90@gmail.com",
  password: "test",
  first_name: "Soumodip",
  last_name: "Sinha",
  balance_points: 500,
  wishlist: [],
  cards_gifted: [],
  cards_received: [],
  socialProfileLink: "testing"
};
export { shallow, mount, render, userDetails, cardDetails };
export default Enzyme;