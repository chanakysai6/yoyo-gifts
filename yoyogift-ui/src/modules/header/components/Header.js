import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import { connect } from 'react-redux'
import { login, logout, createUser } from '../state/actions'
import { bindActionCreators } from 'redux'
import history from '../../common/components/history'
import MySnackBar from '../../common/components/Snackbar'
import Styles from '../../../assets/css/Header.module.css'
import { GoogleLogin, GoogleLogout } from 'react-google-login'
import axiosWrapper from '../../../apis/axiosCreate'

const styles = {
  root: {
    flexGrow: 1,
    flexShrink: 1,
    width: '100%',
  },
  grow: {
    flexGrow: 1,
  },
  toolBar: {
    background: '#32567e',
  },
}

class Header extends Component {
  constructor(props) {
    super(props)
    this.state = {
      showErrorSnack: false,
    }
  }
  render() {
    const { showErrorSnack } = this.state
    const { classes } = this.props
    return (
      <div className={classes.root}>
        {showErrorSnack ? (
          <MySnackBar message="Network Error! Please try again" color="red" />
        ) : null}
        <AppBar position="static">
          <Toolbar className={classes.toolBar}>
            <Typography variant="h6" color="inherit" className={classes.grow}>
              <Button onClick={this.goTolanding}>
                <span style={{ fontSize: '1.2em', color: '#ffffff' }}>
                  YOYOGift
                </span>
                {/* <img
                  src="https://dh3d0az4bd2em.cloudfront.net/yoyogifts/media/lights_logo.png"
                  alt="YOYOGIFTS"
                /> */}
              </Button>
            </Typography>
            {/* {this.props.isLoggedIn ? <Button color="inherit" onClick={this.addUpdateForm}>ADD UPDATE FORM</Button> : null} */}
            {this.props.isLoggedIn ? (
              <Button
                className={Styles.headerButton}
                color="inherit"
                onClick={this.giftsReceived}
              >
                GIFTS RECEIVED
              </Button>
            ) : null}
            {this.props.isLoggedIn ? (
              <Button
                className={Styles.headerButton}
                color="inherit"
                onClick={this.giftsSend}
              >
                GIFTS SENT
              </Button>
            ) : null}
            {this.props.isLoggedIn ? (
              <Button
                className={Styles.headerButton}
                color="inherit"
                onClick={this.myProfile}
              >
                MY PROFILE
              </Button>
            ) : null}
            {/* <Button
              className={Styles.headerButton}
              color="inherit"
              onClick={() => {}}
            >
              {this.props.isLoggedIn ? 'LOGOUT' : 'LOGIN'}
            </Button> */}
            {!this.props.isLoggedIn ? (
              <GoogleLogin
                dataTest="nativeLogin"
                className="GoogButton"
                clientId="813755426604-c14fjkrfta5up8p97rptleprf7ua6l3l.apps.googleusercontent.com"
                buttonText="Login"
                cookiePolicy={'single_host_origin'}
                onSuccess={(response) => {
                  this.googleSuccessResponse(response)
                }}
                onFailure={(response) => {
                  this.googleFailureResponse(response)
                }}
              />
            ) : (
              <GoogleLogout
                buttonText="Logout"
                onLogoutSuccess={(response) => {
                  this.logOut(response)
                }}
              />
            )}
          </Toolbar>
        </AppBar>
      </div>
    )
  }
  googleSuccessResponse = async (response) => {
    const id = response.profileObj.googleId
    await axiosWrapper
      .get(`/users/${id}`)
      .then((profileObj) => {
        window.localStorage.setItem('user', JSON.stringify(profileObj))
        return this.props.login(profileObj)
      })
      .catch(async (err) => {
        const user = await this.props.createUser({
          ...response.profileObj,
          id,
          balance_points: 5000,
        })
        window.localStorage.setItem('user', JSON.stringify(user))
        this.props.login(user)
      })
  }

  googleFailureResponse = (response) => {}

  goTolanding = () => {
    history.push('/')
  }
  myProfile = () => {
    history.push('/Profile')
  }

  giftsSend = () => {
    history.push('/GiftsSend')
  }

  giftsReceived = () => {
    history.push('/GiftsReceived')
  }

  logOut = () => {
    this.props.logout()

    // window.localStorage.removeItem('user')
    // window.sessionStorage.removeItem('user')
    // window.sessionStorage.removeItem('usertype')
    history.push('/')
  }
}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
}

const mapStateToProps = (state) => {
  return {
    isLoggedIn: state.login.loginStatus,
    userDetails: state.login.detailsObject,
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ login, logout, createUser }, dispatch)
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(styles)(Header))
