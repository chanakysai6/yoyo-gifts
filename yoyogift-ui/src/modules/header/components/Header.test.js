import React from 'react';
// import {render} from '@testing-library/react'
import 
 Enzyme,
{shallow} from 'enzyme';
import { Provider } from "react-redux";
// import configureMockStore from "redux-mock-store";
// import { Route } from 'react-router';
import Adapter from 'enzyme-adapter-react-16';
// import { adminEmail } from "../../config/constants";
import {  userDetails } from "../../../Enzyme";

import axiosWrapper from "../../../apis/axiosCreate";
import Header from './Header';
Enzyme.configure({adapter: new Adapter()})

//const mockStore = configureMockStore();
const store = jest.fn();

it('login check with right data',()=>{
const wrapper = shallow(<Provider store={store}><Header /></Provider>  );
// wrapper.find('input[type="text"]').simulate('change', {target: {name: 'username', value: 'krishankantsinghal'}});
// wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: 'krishankant123'}});
// wrapper.find('button').simulate('click');
// expect(wrapper.state('isLogined')).toBe(true);
expect(wrapper).toMatchSnapshot();
})

 it("Native Login, valid email (social media login)", async () => {
    let wrapper = shallow((<button onClick={store}>Ok!</button>)  );
    const loginButton = wrapper.find('button');
    axiosWrapper.get = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        data: [{ ...userDetails, password: undefined }]
      });
    });
    await loginButton.simulate("click");
    wrapper = wrapper.update();
   expect(store.mock.calls.length).toEqual(1);
  });