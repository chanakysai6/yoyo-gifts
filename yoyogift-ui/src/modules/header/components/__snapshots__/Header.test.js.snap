// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login check with right data 1`] = `ShallowWrapper {}`;
